File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        the_map = {} # use hash map
        for i, num in enumerate(nums):
            diff = target - num
            if diff in the_map:
                return [the_map[diff], i]
            
            the_map[num] = i
        
        return
    

File: Data Structures & Algorithms/duplicate-integer/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let theMap = {}
        for (let i = 0; i < nums.length; i++) {
            if (!theMap[nums[i]]) {
                theMap[nums[i]] = 1
            } else {
                return true
            }
        }
        return false
    }
}
